python/tidy/servo_tidy/tidy.py

```diff
@@ -631,28 +631,55 @@ def check_webidl_spec(file_name, contents):
     yield (0, "No specification link found.")
 
 
+class DuplicateKeyError(StandardError):
+    def __init__(self, key):
+        self.key = key
+
+
+class UnsortedKeyError(StandardError):
+    def __init__(self, a, b):
+        self.a = a
+        self.b = b
+
+
 def check_for_possible_duplicate_json_keys(key_value_pairs):
     keys = [x[0] for x in key_value_pairs]
     seen_keys = set()
     for key in keys:
         if key in seen_keys:
-            raise KeyError(key)
+            raise DuplicateKeyError(key)
```

Comment:
Do we really need two error objects just for the sake of raising key errors. Since we're catching the errors later anyway, we could just

``` python
raise KeyError("reason")
```

This way, we also won't be needing any additional attributes later and `yield` the reason directly.